cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/SparkPipelineRunner.java

```diff
@@ -546,7 +546,8 @@ protected SparkCollection<Object> handleJoin(Map<String, SparkCollection<Object>
       Map<String, Schema> inputSchemas = new HashMap<>();
       for (String inputStageName : pipelinePhase.getStageInputs(stageName)) {
         StageSpec inputStageSpec = pipelinePhase.getStage(inputStageName);
-        inputSchemas.put(inputStageName, inputStageSpec.getOutputSchema());
+        inputSchemas.put(inputStageName,
+                         inputStageSpec.getOutputPorts().values().stream().findFirst().orElse(null).getSchema());
```

Comment:
findFirst() isn't the correct thing here, need to fetch the right port based on the stage name.

Reply:
Got it, please see the updated code :)